travis/setup_dependencies_common.sh

```diff
@@ -51,6 +51,41 @@ if [[ $MAIN_CMD == pep8* ]]; then
     return  # no more dependencies needed
 fi
 
+# Pin required versions for dependencies, howto is in FAQ of conda
+# http://conda.pydata.org/docs/faq.html#pinning-packages
+if [[ ! -z $CONDA_DEPENDENCIES ]]; then
+    pin_file=$HOME/miniconda/envs/test/conda-meta/pinned
+    echo $CONDA_DEPENDENCIES | tr " " "\n" | \
+        sed -E -e 's|([a-z]+)([=><!])|\1 \2|g' -e 's| =([0-9])| ==\1|g' > $pin_file
+
+    if [[ $DEBUG == True ]]; then
+        cat $pin_file
+    fi
+
+    # Let env variable version number override this pinned version
+    for package in $(gawk '{print $1}' $pin_file); do
+        version=$(eval echo -e \$$(echo $package | \
+            gawk '{print toupper($0)"_VERSION"}'))
+        if [[ ! -z $version ]]; then
+            gawk -v package=$package -v version=$version \
+                '{if ($1 == package) print package" " version; else print $0}' \
+                $pin_file > /tmp/pin_file_temp
+            mv /tmp/pin_file_temp $pin_file
```

Comment:
It's just bash stupidity, the input and output of the gawk should be different, thus I put everything into a temp file, and after the gawk overwrite the original input with it, then the new ``package`` in the loop starts it from the beginning. So you're right, it's super non-optimal for io but maybe the bash hack is more readable than otherwise it would be.
However this conditional only runs when there is any ``$PACKAGE_VERSION`` variable given, as we said it should override the version numbers listed in ``CONDA_DEPENDENCIES``.